File: FoosballGames.WebApi/PostgreSqlSettings.cs
using Npgsql;

namespace FoosballGames.WebApi;

public class PostgreSqlSettings
{
    public string? Host { get; set; }
    public int Port { get; set; } = 5432;
    public string? Database { get; set; }
    public string? Username { get; set; }
    public string? Password { get; set; }

    public string BuildConnectionString()
    {
        var builder = new NpgsqlConnectionStringBuilder
        {
            Host = Host,
            Port = Port,
            Database = Database,
            Username = Username,
            Password = Password
        };

        return builder.ConnectionString;
    }
}

File: FoosballGames.WebApi/PostgreSqlSettings.fs
namespace FoosballGames.WebApi

open Microsoft.FSharp.Core
open Npgsql

[<CLIMutable>]
type PostgreSqlSettings =
    {Host: string; Port: int; Database: string; Username: string; Password: string}

    member this.BuildConnectionString() =
        let builder =
            NpgsqlConnectionStringBuilder(
                Host = this.Host,
                Port = this.Port,
                Database = this.Database,
                Username = this.Username,
                Password = this.Password
            )

        builder.ConnectionString
